src/services/arweave/WarpDataProvider.ts

```diff
@@ -149,4 +158,87 @@ export class WarpDataProvider implements SmartweaveDataProvider {
       return error;
     }
   }
+
+  async handleTransactionExecution({
+    contractType,
+    interactionType,
+    transactionData,
+  }: {
+    contractType: ContractType;
+    interactionType: AntInteraction | RegistryInteraction;
+    transactionData: any;
+  }): Promise<string | undefined> {
+    try {
+      switch (contractType) {
+        case CONTRACT_TYPES.REGISTRY: {
+          switch (interactionType) {
+            case REGISTRY_INTERACTION_TYPES.BUY_RECORD:
+              {
+                try {
+                  // TODO: implement a function to validate transaction data by interaction type thats takes in interactionType, contractType, and transactionData
+                  if (!transactionData.assetId) {
+                    throw new Error(
+                      'Registry address is required to deploy name purchase',
+                    );
+                  }
+                  const { assetId, ...payload } = transactionData;
+                  const txid = await this.writeTransaction(
+                    new ArweaveTransactionID(assetId),
+                    {
+                      function: 'buyRecord',
```

Comment:
nit, include `function` as a required param of `transactionData`

Reply:
changed